### PR TITLE
Maxquant anndata python conversion

### PR DESCRIPTION
**Pull request by KennethVerheggen**
Created an initial conversion from R to python for maxquant output conversion to anndata

### CODE REVIEW

**Review by rcannood — CHANGES_REQUESTED** [review]
The functionality looks good to me! Happy to see you included a unit tests for the component! :+1: 

Can you take care of the following:

* Add an entry to the changelog
* Run a linter to fix code formatting
* Add type annotations to the functions
* Make the functions return a data frame as opposed to doing inplace replacements

File: src/convert/maxquant_to_h5ad/script.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    "output": "resources_test/zenodo_4274987/maxquant_out/output.h5ad",
+}
+## VIASH END
+
+
+# helper function for transforming column names in proteingroups
+# to snakecase
+def fix_headers(dataframe_old:pd.DataFrame)->pd.DataFrame:
+    """Fixes the headers by unescaping and converting to snakecase and replacing booleans with integers"""
+    dataframe = dataframe_old.copy(deep=True)
```
Comment (DriesSchaumont) [inline src/convert/maxquant_to_h5ad/script.py:17]: Use a for-loop for this: 
```python
from_to = {
    "+": ("and", False),
    "%": ("pct", False),
    " ": ("_", False),
    "[^a-zA-Z0-9_]*", "", regex=True
}
new_columns = dataframe.columns
for old, (new, use_regex) in from_to.items():
    new_columns = new_columns.str.replace(old, new, use_regex=use_regex)
new_columns = new_columns.lower()

```

File: src/convert/maxquant_to_h5ad/script.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    dataframe.columns = dataframe.columns.str.replace("+", "and", regex=False)
+    dataframe.columns = dataframe.columns.str.replace("%", "pct", regex=False)
+    dataframe.columns = dataframe.columns.str.replace(" ", "_", regex=False)
+    dataframe.columns = dataframe.columns.str.lower()
+    dataframe.columns = dataframe.columns.str.replace("[^a-z0-9_]*",
+                                                      "", regex=True)
+    #TODO figure out which are causing the issues
+    for column in dataframe.columns:
```
Comment (DriesSchaumont) [inline src/convert/maxquant_to_h5ad/script.py:25]: pandas has a function to iterate columns:
```python3
for column_name, column in dataframe.items():
    column.replace([False, True], [0, 1], inplace=True)
```

File: src/convert/maxquant_to_h5ad/script.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    for sample_id in sample_ids:
+        headers.append(template.format(sample_id=sample_id))
+    dataframe = _protein_groups.loc[:, _protein_groups.columns.isin(headers)]
+    dataframe.columns = sample_ids
+    dataframe = dataframe.transpose()
+    return dataframe
+
+# read sample metadata
+summary = pd.read_table(f"{par['input']}/combined/txt/summary.txt")
+summary_nt = summary[summary["Raw file"].str.contains("Total") == False]
```
Comment (DriesSchaumont) [inline src/convert/maxquant_to_h5ad/script.py:42]: Does 
```summary[! summary["Raw file"].str.contains("Total")]```
or 
```summary[not summary["Raw file"].str.contains("Total")]```
work?
If so, I would replace with one of these options

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: src/convert/maxquant_to_h5ad/script.py (hunk 1 above)
Comment (KennethVerheggen) [reply src/convert/maxquant_to_h5ad/script.py:17]: Adapted to use a for loop

File: src/convert/maxquant_to_h5ad/script.py (hunk 2 above)
Comment (KennethVerheggen) [reply src/convert/maxquant_to_h5ad/script.py]: Adapted to use the pandas iterator

File: src/convert/maxquant_to_h5ad/script.py (hunk 3 above)
Comment (KennethVerheggen) [reply src/convert/maxquant_to_h5ad/script.py:51]: Tried with various options, none work save for the original one thus far...

**Pull request merged**